### DISCUSSION TITLE
Prisma and encrypted fields

### DISCUSSION THREAD

**Discussion by lkedmi** [prisma/prisma#8107] [Q&A] [answered]
We're considering to use Prisma but not sure if it's a good match for our use-case - for added security, we're doing some column level encryption.

Is there a way to work with encrypted columns with Prisma?

Thanks!

**Comment by ryands17** [comment, marked as answer]
@lkedmi 👋 

I would suggest having a look at Prisma's [Middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware). This would help you encrypt the specific columns you need before adding it to the database.

Let me know your thoughts on this.

**Reply by lkedmi** [reply]
Thanks Ryan, that's definitely helpful.

I don't mind "hacking" it through myself but I hoped someone already did it before since it's a pretty common requirement.

**Reply by ryands17** [reply]
I haven't seen this though. Either client side encryption is performed for E2E encryption or the hosted database's inbuilt encryption feature is used in most cases.

**Reply by VinayaSathyanarayana** [reply]
Example code using Middleware would be most welcome

**Reply by poberbeck** [reply]
> Example code using Middleware would be most welcome

Maybe take a look [here](https://github.com/prisma/prisma/issues/9879#issuecomment-1016285816).

**Comment by idrm** [comment]
[This package](https://github.com/47ng/prisma-field-encryption) may be of interest as well.

**Comment by ACPK** [comment]
Does Prisma's documentation have a list of third-party extensions (i.e. packages)?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
